internal/controller/controllers/cluster_controller.go

```diff
@@ -171,17 +228,121 @@ func (r *ClusterReconciler) updateStatus(ctx context.Context, cluster *adiiov1al
 	SetTimeIfNotNill(&cluster.Status.LastUpdateTime, c.UpdatedAt)
 	SetTimeIfNotNill(&cluster.Status.ControllerLogsCollectionTime, c.ControllerLogsCollectedAt)
 	cluster.Status.ID = c.ID.String()
+	cluster.Status.Error = ""
+}
+
+func (r *ClusterReconciler) updateState(ctx context.Context, cluster *adiiov1alpha1.Cluster, c *common.Cluster) (ctrl.Result, error) {
+	r.syncClusterState(cluster, c)
 
 	if err := r.Status().Update(ctx, cluster); err != nil {
 		r.Log.WithError(err).Error("failed to update error status")
-		return ctrl.Result{}, errors.Wrap(err, "failed to create cluster")
+		return ctrl.Result{Requeue: true}, nil
 	}
 
-	return ctrl.Result{}, nil
+	r.Log.Infof("Updated cluster %s %s state", cluster.Name, cluster.Namespace)
+	return ctrl.Result{RequeueAfter: defaultRequeueClusterDuration}, nil
+}
+
+func (r *ClusterReconciler) updateIfNeeded(ctx context.Context, cluster *adiiov1alpha1.Cluster, c *common.Cluster) (bool, ctrl.Result, error) {
+	update := false
+
+	params := &models.ClusterUpdateParams{}
+
+	spec := cluster.Spec
+
+	updateString := func(new, old string, target **string) {
+		if new != old {
+			*target = swag.String(new)
+			update = true
+			r.Log.Infof("updating %s != %s", old, new)
+		}
+	}
+
+	updateString(spec.Name, c.Name, &params.Name)
+
+	if spec.OpenshiftVersion != c.OpenshiftVersion {
+		return false, ctrl.Result{}, errors.Errorf("Openshift version cannot be updated")
```

Comment:
Do we have a convention regarding capitalizing errors and logs? I don't think it should be capitalized 
https://github.com/golang/go/wiki/CodeReviewComments#error-strings